integrations/kube-agent-updater/pkg/version/teleportping.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/url"
+
+	"github.com/gravitational/trace"
+	"golang.org/x/mod/semver"
+
+	"github.com/gravitational/teleport/integrations/kube-agent-updater/pkg/cache"
+)
+
+const pingPath = "/webapi/ping"
+
+// pingVersionClient retrieves the version from a Teleport ping endpoint.
+// It should not be invoked immediately and must be wrapped in a cache layer in
+// order to avoid spamming the version server in case of reconciliation errors.
+// use PingVersionGetter if you need to get a version.
+type pingVersionClient struct {
+	baseURL *url.URL
+	client  *http.Client
+}
+
+type pingVersionResponse struct {
+	ServerVersion string `json:"server_version"`
+}
+
+// Get sends an HTTP GET request to the webapi/ping route. It expects to receive
+// a JSON containing at least the "server_version" key and whose value follows
+// semver without the "v" prefix.
+func (b *pingVersionClient) Get(ctx context.Context) (string, error) {
+	versionURL := b.baseURL.JoinPath(pingPath)
```

Comment:
we have a client that does this already https://github.com/gravitational/teleport/blob/eadbe9b2a5ee3c1dcc26a6d4b8e2d41e88010e3f/api/client/webclient/webclient.go#L196

Reply:
Yeah
I think we'll need a couple more things to do this request
We might need to add the cluster CA to the trust store (unless it's already part of the system's trust chain)

Reply:
I don't think using the teleport ping client is adequate here for 2 reasons:
- the `teleport-kube-agent-updater` does not depend on `teleport` and adding this dependency will add version constraints and will couple the updater with Teleport more than needed. The updater was designed to be independent from Teleport: it should be able to outlive multiple major versions without being updated. (the rule of thumb is at least 4 major versions)
- the teleport webclient is tied to a Teleport version and might break once the client version and the server version diverge too much. Using the webclient means the updater will not survive changes of the ping response message. Unmarshalling only the version field seems more robust as it will only break if we were to rename or remove the field and we can ignore everything else going on.

>  We might need to add the cluster CA to the trust store (unless it's already part of the system's trust chain)

That's a good point, I designed this to be used against a proxy with trusted certificates. As this is a bonus feature (not required by the RFD but was a low-hanging fruit), adding custom CA support would represent too much work and I'd prefer tp drop the feature entirely. However, we can still trust CA certificates using `SSL_CERT_FILE` and `SSL_CERT_DIR`.

If you think the feature requires more work, I don't have issues with removing the ping version getter entirely. We can bring back the feature later.

Reply:
I agree that importing teleport here just for the web client seems like overkill but I also agree that we probably don't really need this "ping version" client because it's not actually going to be used. Can we just remove it for now?

Reply:
> Unmarshalling only the version field seems more robust as it will only break if we were to rename or remove the field and we can ignore everything else going on.

Should we include comments to that effect on the version field where that response is defined so that folks know that changing the format/name of that field will have breaking affects on this package?

Reply:
I removed the teleportPing so this is no longer an issue :)